### PR TITLE
expose *types.receipt when send transaction

### PR DESCRIPTION
**Pull request by sulenn**

### CODE REVIEW

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- .ci/hello/HelloWorld.go  7
         

Complexity decreasing per file
==============================
+ precompiled/config/system_config_service.go  -1
+ .ci/ethPrecompiled/bn256.go  -4
+ precompiled/crud/crud_service.go  -1
+ .ci/ethPrecompiled/bn256_gm.go  -4
+ precompiled/cns/cns_service.go  -1
         

Clones added
============
- precompiled/chaingovernance/chain_governance_service.go  1
- precompiled/contractlifecycle/contract_life_cycle.go  1
- precompiled/permission/permission.go  3
- precompiled/chaingovernance/chain_governance.go  11
- precompiled/contractlifecycle/contract_life_cycle_service.go  1
- precompiled/crud/table_factory.go  1
- precompiled/consensus/consensus_service.go  1
- .ci/ethPrecompiled/bn256.go  4
- .ci/hello/HelloWorld.go  7
- .ci/ethPrecompiled/bn256_gm.go  4
- precompiled/cns/cns_service.go  1
- precompiled/permission/permission_service.go  1
         

Clones removed
==============
+ precompiled/cns/cns.go  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=5899954&bid=19320259)

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- .ci/hello/HelloWorld.go  7
         

Complexity decreasing per file
==============================
+ precompiled/config/system_config_service.go  -1
+ .ci/ethPrecompiled/bn256.go  -4
+ .ci/ethPrecompiled/bn256_gm.go  -4
+ precompiled/cns/cns_service.go  -1
+ precompiled/crud/crud_service.go  -1
         

Clones added
============
- .ci/ethPrecompiled/bn256.go  4
- .ci/ethPrecompiled/bn256_gm.go  4
- precompiled/consensus/consensus_service.go  1
- precompiled/cns/cns_service.go  1
- precompiled/permission/permission_service.go  1
- precompiled/contractlifecycle/contract_life_cycle.go  1
- precompiled/permission/permission.go  3
- precompiled/chaingovernance/chain_governance_service.go  1
- .ci/hello/HelloWorld.go  7
- precompiled/crud/table_factory.go  1
- precompiled/contractlifecycle/contract_life_cycle_service.go  1
- precompiled/chaingovernance/chain_governance.go  11
         

Clones removed
==============
+ precompiled/cns/cns.go  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=5899954&bid=19320259)

File: .ci/hello/HelloWorld.go [hunk 1]
```diff
@@ -28,10 +28,10 @@ var (
...
+const HelloWorldABI = "[{\"constant\":false,\"inputs\":[{\"name\":\"v\",\"type\":\"string\"}],\"name\":\"set\",\"outputs\":[],\"payable\":false,\"stateMutability\":\"nonpayable\",\"type\":\"function\"},{\"constant\":true,\"inputs\":[],\"name\":\"version\",\"outputs\":[{\"name\":\"\",\"type\":\"string\"}],\"payable\":false,\"stateMutability\":\"view\",\"type\":\"function\"},{\"constant\":true,\"inputs\":[],\"name\":\"get\",\"outputs\":[{\"name\":\"\",\"type\":\"string\"}],\"payable\":false,\"stateMutability\":\"view\",\"type\":\"function\"},{\"inputs\":[],\"payable\":false,\"stateMutability\":\"nonpayable\",\"type\":\"constructor\"},{\"anonymous\":false,\"inputs\":[{\"indexed\":false,\"name\":\"\",\"type\":\"string\"}],\"name\":\"setValue\",\"type\":\"event\"}]"
```
Comment (bxq2011hust) [inline .ci/hello/HelloWorld.go:31]: helloworld.go can be deleted?

File: client/api_handler.go [hunk 2]
```diff
@@ -23,10 +23,10 @@ import (
 
 	"github.com/FISCO-BCOS/go-sdk/conn"
 	"github.com/FISCO-BCOS/go-sdk/core/types"
+	"github.com/bitherhq/go-bither/rlp"
```
Comment (bxq2011hust) [inline client/api_handler.go:26]: "github.com/ethereum/go-ethereum/rlp"

Comment (bxq2011hust) [inline client/api_handler.go:26]:
```suggestion
"github.com/ethereum/go-ethereum/rlp"
```

File: conn/connection.go [hunk 3]
```diff
@@ -624,3 +624,8 @@ func (c *Connection) read(codec ServerCodec) {
 		c.readOp <- readOp{msgs, batch}
 	}
 }
+
+// GetHttp returns whether is HTTP
+func (c *Connection) GetHTTP() bool {
```
Comment (bxq2011hust) [inline conn/connection.go:629]:
```suggestion
func (c *Connection) IsHTTP() bool {
```

File: precompiled/permission/permission_service.go [hunk 4]
```diff
@@ -240,12 +240,7 @@ func (service *Service) list(tableName string) ([]Info, error) {
 	return results, nil
 }
 
-func handleReceipt(c *client.Client, tx *types.Transaction, name string) (int64, error) {
-	// wait for the mining
-	receipt, err := c.WaitMined(tx)
-	if err != nil {
-		return precompiled.DefaultErrorCode, fmt.Errorf("PermissionService wait for the transaction receipt failed, err: %v", err)
-	}
+func handleReceipt(receipt *types.Receipt, name string) (int64, error) {
```
Comment (bxq2011hust) [inline precompiled/permission/permission_service.go:243]:
```suggestion
func parseReturnValue(receipt *types.Receipt, name string) (int64, error) {
```

File: .ci/hello/HelloWorld.go (hunk 1 above)
Comment (sulenn) [reply .ci/hello/HelloWorld.go:31]: 不能，因为有好几个文件的测试需要用到它来创建合约。比如冻结合约，解冻合约。

Comment (bxq2011hust) [reply .ci/hello/HelloWorld.go:31]: ok

**Comment by fisco-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- .ci/hello/HelloWorld.go  7
         

Complexity decreasing per file
==============================
+ .ci/ethPrecompiled/bn256.go  -4
+ precompiled/crud/crud_service.go  -1
+ precompiled/config/system_config_service.go  -1
+ precompiled/cns/cns_service.go  -1
+ .ci/ethPrecompiled/bn256_gm.go  -4
         

Clones added
============
- precompiled/contractlifecycle/contract_life_cycle.go  1
- .ci/ethPrecompiled/bn256.go  4
- .ci/hello/HelloWorld.go  7
- precompiled/consensus/consensus_service.go  1
- precompiled/contractlifecycle/contract_life_cycle_service.go  1
- precompiled/permission/permission.go  3
- precompiled/chaingovernance/chain_governance.go  11
- precompiled/cns/cns_service.go  1
- precompiled/chaingovernance/chain_governance_service.go  1
- precompiled/permission/permission_service.go  1
- .ci/ethPrecompiled/bn256_gm.go  4
- precompiled/crud/table_factory.go  1
         

Clones removed
==============
+ precompiled/cns/cns.go  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/FISCO-BCOS/go-sdk/pullRequest?prid=5899954&bid=19320259)

**Pull request merged**